src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_UInt32.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class PrimitiveParser2
+    {
+        public static partial class InvariantUtf16
+        {
+            public unsafe static bool TryParseUInt32(char* text, int length, out uint value)
+            {
+                int consumed;
+                var span = new ReadOnlySpan<byte>(text, length);
+                return PrimitiveParser2.TryParseUInt32(span, out value, out consumed, EncodingData.InvariantUtf16);
+            }
+            public unsafe static bool TryParseUInt32(char* text, int length, out uint value, out int consumed)
+            {
+                var span = new ReadOnlySpan<byte>(text, length);
+                return PrimitiveParser2.TryParseUInt32(span, out value, out consumed, EncodingData.InvariantUtf16);
+            }
+            public static bool TryParseUInt32(ReadOnlySpan<byte> text, out uint value)
+            {
+                int consumed;
+                return PrimitiveParser2.TryParseUInt32(text, out value, out consumed, EncodingData.InvariantUtf16);
+            }
+            public static bool TryParseUInt32(ReadOnlySpan<byte> text, out uint value, out int consumed)
```

Comment:
One thing I was wondering is whether we should call this parameter consumed or consumedBytes/Chars. Otherwise it's not super clear if it counts chars or bytes in the UTF16 APIs.